crates/clients/matchmaker/src/client.rs

```diff
@@ -0,0 +1,54 @@
+use std::sync::Arc;
+
+use ethers::{signers::Signer, types::Chain};
+
+use jsonrpsee::core::client::ClientT;
+use jsonrpsee::core::Error as RpcError;
+use jsonrpsee::http_client::{transport::HttpBackend, HttpClient, HttpClientBuilder};
+
+use tower::ServiceBuilder;
+
+use crate::{
+    flashbots_signer::{FlashbotsSigner, FlashbotsSignerLayer},
+    types::{BundleRequest, SendBundleResponse},
+};
+
+/// Matchmaker client to interact with MEV-share
+pub struct Client<S> {
+    /// Underlying HTTP client
+    pub http_client: HttpClient<FlashbotsSigner<S, HttpBackend>>,
+}
+
+impl<S: Signer + Clone + 'static> Client<S> {
+    /// Create a new client with the given signer and chain
+    pub fn new(signer: S, chain: Chain) -> Self {
+        let url = match chain {
+            Chain::Mainnet => "https://relay.flashbots.net:443",
```

Comment:
Might be a good idea to add a constants file for the relay urls and other urls that are common amongst MEV strategies even though the relay url is passed into the flashbots executor as a parameter

Reply:
just a small suggestion but it works well!

Reply:
Thanks! Will pull out into a constants file once these start appearing in the codebase more often